### PR TITLE
feat: Protobufs for Tss Startup

### LINKED ISSUE #14751: TSS Service Startup Implementation
Add startup logic for 
* testing conditions of adopting candidate roster that has been keyed and TSS-Ledger-ID feature flag on.
* state changes for adopting candidate roster, ledger id,  and cleaning up state 

- [x] #16823
- [x] #16845
- [x] https://github.com/hashgraph/hedera-services/issues/16553

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes https://github.com/hashgraph/hedera-services/issues/16823 
Related to #14751 

Protobufs for TssEncryptionKeys and TssStatus

### CODE REVIEW

File: hapi/hedera-protobufs/services/auxiliary/tss/tss_encryption_key.proto
Comment (jsync-swirlds) [inline hapi/hedera-protobufs/services/auxiliary/tss/tss_encryption_key.proto:1]: Indentation is incorrect in these proto files.
It should be 4 spaces per level, not 2 (just like the java code).

File: hapi/hedera-protobufs/block/stream/output/state_changes.proto [hunk 1]
```diff
@@ -64,6 +64,13 @@ import "state/token/token.proto";
 import "state/token/token_relation.proto";
 import "state/platform_state.proto";
 import "timestamp.proto";
+import "auxiliary/tss/tss_encryption_key.proto";
+import "auxiliary/tss/tss_message.proto";
+import "auxiliary/tss/tss_vote.proto";
+import "auxiliary/tss/tss_share_signature.proto";
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hapi/hedera-protobufs/block/stream/output/state_changes.proto:70]: Agreed. Removed. Thanks @jsync-swirlds

File: hapi/hedera-protobufs/services/state/tss/tss_status.proto [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+ * This key SHALL be used to uniquely identify entries in the Message Maps.
+ */
+message TssStatus {
+
+  /**
+   * An enum representing the status of the TSS keying process. This status
+   * SHALL be used to determine the state of the TSS keying process.
+   * <p>
+   * This value MUST be set.<br/>
+   * This value MUST NOT be empty.<br/>
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hapi/hedera-protobufs/services/state/tss/tss_status.proto:47]: Updated. Thank you!

File: hapi/hedera-protobufs/services/state/tss/tss_status.proto [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+   */
+  TssKeyingStatus tss_keying_status = 1;
+
+  /**
+   * An enum representing the key either active roster or candidate roster.
+   * This value will be to key active roster if it is genesis stage
+   * <p>This value MUST be set.<br/>
+   * This value MUST be a valid sequence number.
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hapi/hedera-protobufs/services/state/tss/tss_status.proto:55]: Updated. Thank you!

File: hapi/hedera-protobufs/services/state/tss/tss_status.proto [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+   */
+  RosterToKey roster_to_key = 2;
+
+  /**
+   * A hash of the ledger_id resulted from the TSS keying process.
+   * If this value is empty, the TSS keying process has not yet completed.
+   * <p>
+   * This value MUST be set.<br/>
+   * This value MUST NOT be empty.<br/>
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hapi/hedera-protobufs/services/state/tss/tss_status.proto:64]: Updated the docs.

File: hapi/hedera-protobufs/services/state/tss/tss_status.proto [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+     * This status SHALL be used to determine the state of the TSS keying process.
+     * This value MUST be set when tss is enabled.<br/>
+     * This value MUST be set when tss is enabled.<br/>
+     */
+    TssKeyingStatus tss_keying_status = 1;
+
+    /**
+     * An enum representing the key either active roster or candidate roster.<br/>
+     * This value will be to key active roster if it is genesis stage
+     * <p>This value MUST be set.<br/>
```
Comment (jsync-swirlds) [inline hapi/hedera-protobufs/services/state/tss/tss_status.proto:54]: Another missing newline.
```suggestion
     * <p>
     * This value MUST be set.
```

File: hapi/hedera-protobufs/services/state/tss/tss_status.proto [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+ * This key SHALL be used to determine the stage of the TSS keying process.
+ */
+message TssStatus {
+
+    /**
+     * An enum representing the status of the TSS keying process.<br/>
+     * <p>
+     * This status SHALL be used to determine the state of the TSS keying process.
+     * This value MUST be set when tss is enabled.<br/>
+     * This value MUST be set when tss is enabled.<br/>
```
Comment (jsync-swirlds) [inline hapi/hedera-protobufs/services/state/tss/tss_status.proto:47]: Duplicated line
```suggestion
     * This status SHALL be used to determine the state of the TSS keying process.<br/>
     * This value MUST be set when tss is enabled.
```

**Review by timo0 — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM, tyvm @Neeharika-Sompalli !

File: hapi/hedera-protobufs/block/stream/output/state_changes.proto [hunk 7]
```diff
@@ -857,6 +879,21 @@ message MapChangeValue {
...
+
+        /**
+         * A TSS Encryption Key Transaction Body value.
+         */
+        com.hedera.hapi.services.auxiliary.tss.TssMessageTransactionBody tss_message_value = 21;
+
+        /**
+         * A TSS Encryption Key Transaction Body value.
+         */
+        com.hedera.hapi.services.auxiliary.tss.TssVoteTransactionBody tss_vote_value = 22;
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hapi/hedera-protobufs/block/stream/output/state_changes.proto]: This state is essentially "the history of important exchanged TSS messages", so imo it's not surprising to find that it contains transaction bodies.

[the remark this answers is not in the document]
Comment (jsync-swirlds) [reply hapi/hedera-protobufs/block/stream/output/state_changes.proto]: Storing a history of "important" TSS messages is, in itself a bit odd (we don't store any other transactions in state).
It just seems like we've probably missed something somewhere; but it's not worth holding up the PR for that.

**Pull request merged**